composer/utils/file_helpers.py

```diff
@@ -52,6 +53,36 @@
 ]
 
 
+@retry(num_attempts=5)
+def download_object_or_file(
+    object_store: Union[ObjectStore, LoggerDestination],
+    object_name: str,
+    file_destination: Union[str, pathlib.Path],
+    overwrite: bool = False,
+    progress_bar: bool = False,
+):
+    if isinstance(object_store, ObjectStore):
+        if progress_bar:
+            callback = _get_callback(f'Downloading {object_name}')
+        else:
+            callback = None
+        object_store.download_object(
+            object_name=object_name,
+            filename=file_destination,
+            overwrite=overwrite,
+            callback=callback,
+        )
+    elif isinstance(object_store, LoggerDestination):
+        object_store.download_file(
+            remote_file_name=object_name,
+            destination=str(file_destination),
+            overwrite=overwrite,
+            progress_bar=progress_bar,
+        )
+    else:
+        raise RuntimeError(f'Unrecognized type: {type(object_store)}')
```

Comment:
```suggestion
    else:
        raise RuntimeError(f'Unrecognized type: {type(object_store)} when attempting to download object')
```
nit: a bit more user friendly error?